build/xtask/src/dist.rs

```diff
@@ -1551,24 +1606,32 @@ pub fn make_kconfig(
     // The remaining regions are allocated to tasks on a first-come first-serve
     // basis.
     for (i, (name, task)) in toml.tasks.iter().enumerate() {
-        if power_of_two_required && !task.requires["flash"].is_power_of_two() {
-            panic!("Flash for task '{}' is required to be a power of two, but has size {}", task.name, task.requires["flash"]);
+        if power_of_two_required
+            && !task_allocations[name]["flash"].len().is_power_of_two()
```

Comment:
This is not a new bug, but: these conditions need to be applied to _all_ regions, not just `"flash"` and `"ram"`. For instance the `sram1` area used by the ethernet driver still needs to be pow2.